compilers/concrete-compiler/compiler/lib/Dialect/FHE/Transforms/Optimizer.cpp

```diff
@@ -56,6 +56,19 @@ struct OptimizerPartitionFrontierMaterializationPass
     mlir::func::FuncOp func = this->getOperation();
 
     func.walk([&](mlir::Operation *producer) {
+      mlir::IRRewriter rewriter(producer->getContext());
+
+      // Remove the change_partition op.
+      // TODO: The crypto parameters used in the op should be considered before
+      // removal
+      if (mlir::dyn_cast_or_null<FHELinalg::ChangePartitionEintOp>(producer) ||
+          mlir::dyn_cast_or_null<FHE::ChangePartitionEintOp>(producer)) {
+        rewriter.startRootUpdate(func);
+        rewriter.replaceOp(producer, producer->getOperand(0));
+        rewriter.finalizeRootUpdate(func);
+        return;
+      }
```

Comment:
Maybe put this as a separate pass to be run after the ConcreteOptimizer Analysis